### ISSUE TITLE
Objeck VM built on MSYS2 CLANG64 can't be used to run objeck-lsp in STDIO mode?

### ISSUE THREAD

**Issue by iahung** [objeck/objeck-lang#196]
Noticed the output of it on MSYS2 CLANG64 (https://github.com/objeck/objeck-lang/issues/193), it seems `OBJECK_STDIO=binary` doesn't affect it.

I also noticed it seems it doesn't support Unicode output too, because with or without `OBJECK_STDIO=binary`, it failed to print the last two lines (which contain Unicode characters). It can only print the first line (which doesn't contain Unicode characters).

**Comment by iahung** [comment]
I recall Objeck VM built on MSYS2 CLANG64 used to have proper Unicode output. The program https://github.com/objeck/objeck-lang/blob/master/programs/deploy/hello_0.obs used to print correctly. There must be something changed.

**Comment by objeck** [comment]
I have not idea how to [enable](https://stackoverflow.com/questions/59562534/how-to-change-console-program-for-unicode-support-in-windows) Unicode support in the MSYS2 CLANG64 shell. 

Suggestions?

**Comment by objeck** [comment]
The MSYS2 CLANG64 console is screwed up. Please point me to threads to sort this. It is its own animal and not an issue in macOS.

**Comment by iahung** [comment]
Unicode output is something that always worked for me until this release (maybe it broke since even earlier releases, as recently I only tested building but not running any programs using the deployed Objeck VMs). I build and test my own version on MSYS2, so I know it very well. Even on Cygwin, Unicode output also works. I guess it's broken since support for STDIO was added to objeck-lsp as the Objeck VM also needed to change to support this.

**Comment by objeck** [comment]
There is a compiler directive that ignores MSYS2 CLANG64 for STDIO. This is a shell issue, but I am not sure to code around it.

**Comment by iahung** [comment]
> I have no idea how to [enable](https://stackoverflow.com/questions/59562534/how-to-change-console-program-for-unicode-support-in-windows) Unicode support in the MSYS2 CLANG64 shell.
> 
> Suggestions?



> I have no idea how to [enable](https://stackoverflow.com/questions/59562534/how-to-change-console-program-for-unicode-support-in-windows) Unicode support in the MSYS2 CLANG64 shell.
> 
> Suggestions?

Even on that thread, the OP acknowledged the fault was not MinGW's. The information that Unicode output doesn't work on MSYS2 CLANG64 is wrong. I thought there was no reason for this to be true. I did an experiment by building the Objeck VM on MSYS2 CLANG64 and testing hello_0.obs. It worked. I was able to successfully debunk this information.

**Comment by objeck** [comment]
Not about debunking; please take a look at the code commits. I did not change anything I am aware of that impacted MSYS2 CLANG64. Moreover, I never had Unicode support working with that shell. I will reopen the issue and see if there is a solution. Yes, odd that Unicode support works fine in all other Windows shells; I must be doing some wrong.

**Comment by iahung** [comment]
> Not about debunking; please take a look at the code commits. I did not change anything I am aware of that impacted MSYS2 CLANG64. Moreover, I never had Unicode support working with that shell. I will reopen the issue and see if there is a solution. Yes, odd that Unicode support works fine in all other Windows shells; I must be doing some wrong.

Even if I take a look, I will understand nothing. I'm always tell you I'm a packager (someone build people's code) and not a developer :smile: But if I need to make a suggestion to you, I will suggest you try removing the workaround for MSYS2 CLANG64 to see if it works. In other words, treat MSYS2 CLANG64 just like MSYS2 UCRT64.

**Comment by iahung** [comment]
> Thanks, I did; it does not. I appreciate your feedback, and I will explore this further.

https://stackoverflow.com/questions/59562534/how-to-change-console-program-for-unicode-support-in-windows

More information: the first sample code on that thread doesn't work as it failed to print the Unicode string, the second sample code on that thread, the UTF-8 string printed correctly, the UTF-16 string didn't.

**Comment by objeck** [comment]
Thanks, it is a very specific use case for on font family. For other shells, I do need to code down to this level.

```
cfi.cbSize = sizeof(CONSOLE_FONT_INFOEX);
cfi.nFont = 0;
cfi.dwFontSize.X = 8;
cfi.dwFontSize.Y = 16;
cfi.FontFamily = FF_SIMHEI;
cfi.FontWeight = FW_NORMAL;
wcscpy(cfi.FaceName, L"SimHei");
```

**Comment by iahung** [comment]
> Thanks, it is a very specific use case for one font family. For other shells, I do not need to code down to this level.
> 
> ```
> cfi.cbSize = sizeof(CONSOLE_FONT_INFOEX);
> cfi.nFont = 0;
> cfi.dwFontSize.X = 8;
> cfi.dwFontSize.Y = 16;
> cfi.FontFamily = FF_SIMHEI;
> cfi.FontWeight = FW_NORMAL;
> wcscpy(cfi.FaceName, L"SimHei");
> ```

Is it something right to do? SimHei is a font specific for Chinese characters. What if you need to output Cyrillic characters? I think it's simpler that we accept Unicode output is broken on MSYS2 CLANG64. Either it's never worked like you think or it's just broken recently like I think, it doesn't matter. Btw, happy hacking.

**Comment by objeck** [comment]
Fixed in af42131d0adcb1af35343c4b81ebc037a05aad34. Going to close as MSYS2 builds are released and the issue is only for MSYS2 Clang 64.
